src/internal/m365/collection/drive/collections_tree_test.go

```diff
@@ -0,0 +1,521 @@
+	}
+
+	mdColl, err := graph.MakeMetadataCollection(
+		pathPrefix,
+		[]graph.MetadataCollectionEntry{
+			graph.NewMetadataEntry(bupMD.DeltaURLsFileName, prevDeltas),
+			graph.NewMetadataEntry(bupMD.PreviousPathFileName, previousPaths),
+		},
+		func(*support.ControllerOperationStatus) {},
+		count.New())
+	require.NoError(t, err, "creating metadata collection", clues.ToCore(err))
+
+	return []data.RestoreCollection{
+		dataMock.NewUnversionedRestoreCollection(t, data.NoFetchRestoreCollection{Collection: mdColl}),
+	}
+}
+
+func compareMetadata(
+	t *testing.T,
+	mdColl data.Collection,
+	expectDeltas map[string]string,
+	expectPrevPaths map[string]map[string]string,
+) {
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	colls := []data.RestoreCollection{
+		dataMock.NewUnversionedRestoreCollection(t, data.NoFetchRestoreCollection{Collection: mdColl}),
+	}
+
+	deltas, prevs, _, err := deserializeAndValidateMetadata(
+		ctx,
+		colls,
+		count.New(),
+		fault.New(true))
+	require.NoError(t, err, "deserializing metadata", clues.ToCore(err))
+	assert.Equal(t, expectDeltas, deltas, "delta urls")
+	assert.Equal(t, expectPrevPaths, prevs, "previous paths")
+}
+
+// for comparisons done by collection state
+type stateAssertion struct {
+	itemIDs []string
+	// should never get set by the user.
+	// this flag gets flipped when calling assertions.compare.
+	// any unseen collection will error on requireNoUnseenCollections
+	sawCollection bool
+}
+
+// for comparisons done by a given collection path
+type collectionAssertion struct {
+	doNotMerge    assert.BoolAssertionFunc
+	states        map[data.CollectionState]*stateAssertion
+	excludedItems map[string]struct{}
+}
+
+type statesToItemIDs map[data.CollectionState][]string
+
+func newCollAssertion(
+	doNotMerge bool,
+	itemsByState statesToItemIDs,
+	excludeItems ...string,
+) collectionAssertion {
+	states := map[data.CollectionState]*stateAssertion{}
+
+	for state, itemIDs := range itemsByState {
+		states[state] = &stateAssertion{
+			itemIDs: itemIDs,
+		}
+	}
+
+	dnm := assert.False
+	if doNotMerge {
+		dnm = assert.True
+	}
+
+	return collectionAssertion{
+		doNotMerge:    dnm,
+		states:        states,
+		excludedItems: makeExcludeMap(excludeItems...),
+	}
+}
+
+// to aggregate all collection-related expectations in the backup
+// map collection path -> collection state -> assertion
+type collectionAssertions map[string]collectionAssertion
+
+// ensure the provided collection matches expectations as set by the test.
+func (cas collectionAssertions) compare(
+	t *testing.T,
+	coll data.BackupCollection,
+	excludes *prefixmatcher.StringSetMatchBuilder,
+) {
+	ctx, flush := tester.NewContext(t)
+	defer flush()
+
+	var (
+		itemCh  = coll.Items(ctx, fault.New(true))
+		itemIDs = []string{}
+	)
+
+	var p path.Path
+
+	switch coll.State() {
+	case data.DeletedState:
+		p = coll.PreviousPath()
+	default:
+		p = coll.FullPath()
+	}
+
+	for {
+		itm, ok := <-itemCh
+		if !ok {
+			break
+		}
+
```

Comment:
nit: range over the item channel so channel close doesn't have to be handled separately
```suggestion
	for itm := range <-itemCh {
```

Reply:
I feel like I should have known that already 🤔